src/ai/ai.cpp

```diff
@@ -2735,6 +2800,14 @@ bool will_accept_peace_offer(sys::state& state, dcon::nation_id n, dcon::nation_
 		if(war_duration < 365) {
 			return false;
 		}
+		//The country is ruined and it's been a bit into the war, we'll recover later
+		if(state.world.nation_get_war_exhaustion(n) > 80) {
+			return true;
+		}
+		//Stalemated war, peace out
+		if(war_duration > (365 * 6) && (overall_score <= 10 && overall_score >= -10)) {
```

Comment:
maybe only if the offer is a white peace? Otherwise this will let you take 100% warscore in a stalemated war